From 76358fe6d4fc1f4db658bec9a36ccb52e946fafd Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 15 Apr 2024 10:54:36 +0200
Subject: [PATCH 0466/1695] scpfw: arch/none/optee: Define OP-TEE passed CMake
 config switches

Define CMake CFG_SCPFW_MOD_* confi switches set by OP-TEE for
scp-firmware configuration. This change prevents build warning
traces like the below one:

CMake Warning:
  Manually-specified variables were not used by the project:

    CFG_SCPFW_MOD_DVFS
    CFG_SCPFW_MOD_MOCK_CLOCK
    CFG_SCPFW_MOD_MOCK_PPU
    CFG_SCPFW_MOD_MOCK_PSU
    CFG_SCPFW_MOD_MSG_SMT
    CFG_SCPFW_MOD_OPTEE_CONSOLE

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git aa71ac552c56f876b7644ca5a8df9c235a2dbad4 scpfw: arch/none/optee: Define OP-TEE passed CMake config switches]
Change-Id: I9e04005ad56326b872d900cb5809fe3f95dd185a
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 .../arch/none/optee/CMakeLists.txt            | 30 +++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt
index 8caf11be9..1cb21fac8 100644
--- a/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt
@@ -5,6 +5,36 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
+# OP-TEE CMake switches
+set (CFG_SCPFW_MOD_DVFS n)
+set (CFG_SCPFW_MOD_CLOCK n)
+set (CFG_SCPFW_MOD_MOCK_CLOCK n)
+set (CFG_SCPFW_MOD_MOCK_PPU n)
+set (CFG_SCPFW_MOD_MOCK_PSU n)
+set (CFG_SCPFW_MOD_MSG_SMT n)
+set (CFG_SCPFW_MOD_OPTEE_CONSOLE n)
+set (CFG_SCPFW_MOD_OPTEE_CLOCK n)
+set (CFG_SCPFW_MOD_OPTEE_MBX n)
+set (CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR n)
+set (CFG_SCPFW_MOD_OPTEE_RESET n)
+set (CFG_SCPFW_MOD_OPTEE_SMT n)
+set (CFG_SCPFW_MOD_OPTEE_VOLTD_REGULATOR n)
+set (CFG_SCPFW_MOD_POWER_DOMAIN n)
+set (CFG_SCPFW_MOD_PSU n)
+set (CFG_SCPFW_MOD_REG_SENSOR n)
+set (CFG_SCPFW_MOD_RESET_DOMAIN n)
+set (CFG_SCPFW_MOD_SCMI n)
+set (CFG_SCPFW_MOD_SCMI_APCORE n)
+set (CFG_SCPFW_MOD_SCMI_CLOCK n)
+set (CFG_SCPFW_MOD_SCMI_PERF n)
+set (CFG_SCPFW_MOD_SCMI_POWER_DOMAIN n)
+set (CFG_SCPFW_MOD_SCMI_RESET_DOMAIN n)
+set (CFG_SCPFW_MOD_SCMI_SENSOR n)
+set (CFG_SCPFW_MOD_SCMI_VOLTAGE_DOMAIN n)
+set (CFG_SCPFW_MOD_SENSOR n)
+set (CFG_SCPFW_MOD_SYSTEM_PLL n)
+set (CFG_SCPFW_MOD_VOLTAGE_DOMAIN n)
+
 add_library(arch-optee)
 
 if(NOT CFG_NUM_THREADS)
-- 
2.47.2

